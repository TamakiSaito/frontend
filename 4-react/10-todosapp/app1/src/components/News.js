/* eslint-disable no-undef */
import React from "react";
import { useState, useEffect } from "react";
import "./News.css";

export default function News() {
  const [news, setNews] = useState([]);

  // mounted
  useEffect(() => {
    fetch("https://api.hnpwa.com/v0/news.json")
      .then((res) => res.json())
      .then((json) => {
        setNews(json);
      });
  }, []);

  console.log("news: ", news);

  return (
    <div>
      <h1>News</h1>
      <div>{render}</div>
    </div>
  );
}
