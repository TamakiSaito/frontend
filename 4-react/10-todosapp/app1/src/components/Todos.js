import React, { useState, useEffect } from "react";
import "./Todos.css";

export default function Todos() {
  const [todos, setTodos] = useState([]);
  const [inputText, setInputText] = useState("");
  // keyが重複しないように作成
  const [todoKey, setTodoKey] = useState(0);

  // Mounted
  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/todos")
      .then((res) => res.json())
      .then((json) => {
        const result = json.filter((item) => item.useId === 1);
        setTodos(result);
        setTodoKey(result.length + 1);
      });
  }, []);

  const render = todos.map((todo) => {
    return (
      <div key={todo.id}>
        <div>
          <span className="todoTitle" onClick={() => modTodo(todo.id)}>
            #{todo.id} / {todo.title}
          </span>
          <span className="todoCompleted" onClick={() => modTodo(todo.id)}>
            {todo.completed ? "DONE ✅" : "TODO"}
          </span>
          <button className="todoBtn" onClick={() => delTodo(todo.id)}>
            ❌
          </button>
        </div>
      </div>
    );
  });

  // Insert 
  // setTodosに(addItem)を入れて変更させる
  // inputTextにSENDボタンを押したものが入って
  // { id: todoKey, title: inputText }に表示させる
  const addTodo = (inputText) => {
    const addItem = [...todos, { id: todoKey, title: inputText }];
    // 既存のkey + 1
    setTodoKey(todoKey + 1);
    setTodos(addItem);
    // SENDボタンを押したらテキストをクリアする
    setInputText("");
  };

  // Delete
  // もらったidと同じだったら削除
  const delTodo = (id) => {
    const delItem = todos.filter((item) => {
      return item.id !== id;
    });
    setTodos(delItem);
  };

  // Update
  const modTodo = (id) => {
    const modItem = todos.map((item) => {
      return item.id === id ? { ...item, completed: !item.completed } : item;
    });
    setTodos(modItem);
  };

  return (
    <div>
      <h1>Todos</h1>
      <div>
        <input
          type="text"
          placeholder="New Todo"
          value={inputText}
          // inputTextの中身を変更する
          onChange={(e) => {
            setInputText(e.target.value);
          }}
        ></input>
        {/* SENDボタンをクリックしたらinputTextの内容をaddTodoに */}
        <button onClick={() => addTodo(inputText)}>SEND</button>
      </div>
      <div>{render}</div>
    </div>
  );
}
