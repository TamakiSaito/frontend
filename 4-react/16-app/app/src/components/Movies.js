import "./Movie.css";
import React from "react";
import Box from "@mui/material/Box";
import Rating from "@mui/material/Rating";
import CircularProgress from "@mui/material/CircularProgress";
import Tooltip from "@mui/material/Tooltip";
import Accordion from "@mui/material/Accordion";
import AccordionSummary from "@mui/material/AccordionSummary";
import AccordionDetails from "@mui/material/AccordionDetails";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";

import { useState, useEffect } from "react";

export default function Movies() {
  const [movies, setMovies] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  const movieRatingClass = (rating) => {
    return rating >= 9 ? "good" : rating >= 7 ? "soso" : "bad";
  };

  // mounted
  useEffect(() => {
    // Loading中のフラグ
    setIsLoading(true);
    //fetch("https://yts.mx/api/v2/list_movies.json?sort_by=rating")
    fetch("https://yts.mx/api/v2/list_movies.json")
      .then((res) => res.json())
      .then((json) => {
        setIsLoading(false);
        setMovies(json.data.movies);
      });
  }, []);

  console.log("movies: ", movies);

  const loading = (
    <Box sx={{ display: "flex" }}>
      <CircularProgress />
    </Box>
  );

  const render = movies.map((movie) => {
    return (
      <div key={movie.id}>
        <a className="movieTitle" href={movie.url}>
          {movie.title}({movie.year})
        </a>
        <img
          clasName="movieImg"
          src={movie.large_cover_image}
          alt={movie.title}
        ></img>

        <div className="detail">
          <Accordion>
            <AccordionSummary
              expandIcon={<ExpandMoreIcon />}
              aria-controls="panel1-content"
              id="panel1-header"
            >
              Click Here To Read More
            </AccordionSummary>
            <AccordionDetails>
              <div>
                評価:
                <Tooltip title={movie.rating + "/10点"} placement="right">
                  <Box
                    display="inline"
                    sx={{
                      "& > legend": { mt: 2 },
                    }}
                  >
                    <Rating
                      size="small"
                      name="simple-controlled"
                      readOnly
                      value={movie.rating / 2}
                    />
                  </Box>
                </Tooltip>
              </div>
              <div>ジャンル:{movie.gernes}</div>
              <div>ランタイム:{movie.runtime}分</div>
              <div>ストーリー:{movie.synopsis || "情報なし"}</div>
            </AccordionDetails>
          </Accordion>
        </div>
      </div>
    );
  });

  return (
    <div>
      <h1>Movies</h1>
      <div>{isLoading ? loading : render}</div>
    </div>
  );
}
