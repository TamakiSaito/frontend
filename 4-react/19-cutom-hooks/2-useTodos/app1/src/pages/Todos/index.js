import React, { useState } from "react";
import Label from "../../components/Atoms/Label";
import Checkbox from "@mui/material/Checkbox";
import IconButton from "@mui/material/IconButton";
import DeleteIcon from "@mui/icons-material/Delete";
import TextField from "@mui/material/TextField";
import NormalButton from "../../components/Atoms/NormalButton";
import useTodos from "../../composables/useTodos";

export default function Todos() {
  const todos = useTodos();

  const [inputText, setInputText] = useState("");

  // Mounted
  // useEffect(() => {
  //   fetch("https://jsonplaceholder.typicode.com/todos")
  //     .then((res) => res.json())
  //     .then((json) => {
  //       const result = json.filter((item) => item.useId === 1);
  //       setTodos(result);
  //       setTodoKey(result.length + 1);
  //     });
  // }, []);

  const render = todos.todos.map((todo) => {
    return (
      <div key={todo.id}>
        <div>
          <span
            className={
              todo.completed ? "text-gray-500 line-through" : "font-bold"
            }
            onClick={() => toggleTodo(todo.id)}
          >
            #{todo.id} / {todo.title}
          </span>
          <Checkbox
            checked={todo.completed}
            onChange={() => toggleTodo(todo.id)}
          />
          <IconButton aria-label="delete" onClick={() => delTodo(todo.id)}>
            <DeleteIcon />
          </IconButton>
        </div>
      </div>
    );
  });

  // Methods
  const addTodo = (inputText) => {
    todos.addTodo(inputText);
    setInputText("");
  };

  const delTodo = (id) => {
    todos.delTodo(id);
  };

  const toggleTodo = (id) => {
    todos.toggleTodo(id);
  };

  return (
    <div>
      <Label text="Todos" />

      <div className="mb-[20px]">
        <span className="mr-[5px]">
          <TextField
            value={inputText}
            size="small"
            required
            placeholder="New Todo"
            onChange={(e) => {
              setInputText(e.target.value);
            }}
          ></TextField>
        </span>
        <NormalButton buttonText="SEND" onClick={() => addTodo(inputText)} />
      </div>
      <div>{render}</div>
    </div>
  );
}
