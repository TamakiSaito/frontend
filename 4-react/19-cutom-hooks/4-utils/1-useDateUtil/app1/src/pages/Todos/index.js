import React, { useState } from "react";
import Label from "../../components/Atoms/Label";
import Checkbox from "@mui/material/Checkbox";
import IconButton from "@mui/material/IconButton";
import DeleteIcon from "@mui/icons-material/Delete";
import TextField from "@mui/material/TextField";
import NormalButton from "../../components/Atoms/NormalButton";
import useTodos from "../../composables/useTodos";
import useDateUtil from "../../utils/useDateUtil";

export default function Todos() {
  const todos = useTodos();

  const [inputText, setInputText] = useState("");

  // Mounted
  // useEffect(() => {
  //   fetch("https://jsonplaceholder.typicode.com/todos")
  //     .then((res) => res.json())
  //     .then((json) => {
  //       const result = json.filter((item) => item.useId === 1);
  //       setTodos(result);
  //       setTodoKey(result.length + 1);
  //     });
  // }, []);

  const render = todos.todos.map((todo) => {
    return (
      <div key={todo.id}>
        <div>
          <span
            className={
              todo.completed ? "text-gray-500 line-through" : "font-bold"
            }
            onClick={() => toggleTodo(todo.id)}
          >
            #{todo.id} / {todo.title}
          </span>
          <Checkbox
            checked={todo.completed}
            onChange={() => toggleTodo(todo.id)}
          />
          <IconButton aria-label="delete" onClick={() => delTodo(todo.id)}>
            <DeleteIcon />
          </IconButton>
        </div>
      </div>
    );
  });

  // Methods
  const addTodo = (inputText) => {
    todos.addTodo(inputText);
    setInputText("");
  };

  const delTodo = (id) => {
    todos.delTodo(id);
  };

  const toggleTodo = (id) => {
    todos.toggleTodo(id);
  };

  const dateUtil = useDateUtil();
  const formattedDate = dateUtil.formatDate(
    dateUtil.currentDate,
    "YYYY-MM-DD HH:mm"
  );

  return (
    <div>
      <Label text="Todos" />

      <div className="mb-[20px]">
        <span className="mr-[5px]">
          <TextField
            value={inputText}
            size="small"
            required
            placeholder="New Todo"
            onChange={(e) => {
              setInputText(e.target.value);
            }}
          ></TextField>
        </span>
        <NormalButton buttonText="SEND" onClick={() => addTodo(inputText)} />
      </div>

      <p>現在の時刻: {formattedDate}</p>
      <div>{render}</div>
    </div>
  );
}
