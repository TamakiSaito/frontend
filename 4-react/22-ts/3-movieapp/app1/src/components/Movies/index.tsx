import { MoviesProps } from '../../common/types'
import './styles.css'

const Movies: React.FC<MoviesProps> = ({ movies }) => {
  const changeToHHMM = (mins: number) => {
    let h: number | string = Math.floor(mins / 60);
    let m: number | string = mins % 60;
    return `${h}h ${m}min`;  
  }
  
  return (
    <div>
      {movies.map((movie, index) => (
        <div key={index}>
          <img src={movie.large_cover_image} alt={movie.title} />
          <div>{movie.title}({movie.year})</div>
          <div className={movie.rating >= 9 ? 'good' : movie.rating >= 7 ? 'soso' : 'bad' }>
            Rating: {movie.rating} / 10
          </div>
          <div>Genres: {movie.genres.join(", ")}</div>
          <div>Runtime: {changeToHHMM(movie.runtime)}</div>
        </div>
      ))}
    </div>
  )
}

export default Movies;
