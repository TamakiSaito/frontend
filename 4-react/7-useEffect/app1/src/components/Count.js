// Count.jsを開いた時にuseEffect
import { useState, useEffect } from "react";


export default function Count() {
  //let count = 0の代わりに
  const [count, setCount] =useState(0);


  const increase = () => {
    setCount(count + 1);
  };

  const decrease = () => {
    setCount(count - 1);
  };

  // Lifecycle Hooks

  // mounted (returnなし)：countを押した時
  useEffect(() => {
    console.log("mounted");
  }, []);

  // updated : countの数が変わって実行された時
  useEffect(() => {
    console.log("updated:", count);
  },[count]);

  //unmounted (returnあり)：再びcountを押した時
  useEffect(() => {
    return () => {
      console.log("unmounted");
    };
  }, []);

  return ( 
    <div className="App">
      <div>count : {count}</div>
      <button onClick={increase}>+</button>
      <button onClick={decrease}>-</button>
    </div>
  );
}
