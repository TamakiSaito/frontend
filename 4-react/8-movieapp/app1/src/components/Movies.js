import React from "react";
import { useState, useEffect } from "react";
import "./Movie.css";

export default function Movies() {
  const [movies, setMovies] = useState([]);

  // mounted
  useEffect(() => {
    fetch("https://yts.mx/api/v2/list_movies.json")
    // responseをもらって、json化して、json化したものがmoviesに入る
    .then((res) => res.json())
    .then((json) => { 
      setMovies(json.data.movies);
     });
  },[]);

  console.log("movies: ", movies);

  // moviesの中のtitleをmapで取得
  const render = movies.map(movie => {
    return (
      <div key={movie.id}>
        <a className="movieTitle" href={movie.url}>{movie.title}</a>
        <img clasName="movieImg" src={movie.large_cover_image} alt={movie.title}></img>
      </div>
    );
  });

  return (
  <div>
    <h1>Movies</h1>
    <div>{render}</div>
    
  </div>
  );
}
