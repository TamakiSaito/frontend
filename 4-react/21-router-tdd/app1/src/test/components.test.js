import React from "react";
import { render, fireEvent, screen, waitFor, act } from "@testing-library/react";

import "@testing-library/jest-dom";
import Count from "../components/Count";
import Movies from "../components/Movies";
import News from "../components/News";
import Todos from "../components/Todos";

describe("<Count />のレンダリングテスト", () => {
  it("Snapshot test", () => {
    const utils = render(<Count />);
    expect(utils.container).toMatchSnapshot();
  });

  it("初期カウントが0で表示されること", () => {
    const { getByText } = render(<Count />);
    expect(getByText("Count 0")).toBeInTheDocument();
  });

  it("+ボタンをクリックするとカウントが1増えること", () => {
    const { getByText } = render(<Count />);
    fireEvent.click(getByText("+"));
    expect(getByText("Count 1")).toBeInTheDocument();
  });

  it("-ボタンをクリックするとカウントが1減ること", () => {
    const { getByText } = render(<Count />);
    fireEvent.click(getByText("-"));
    expect(getByText("Count -1")).toBeInTheDocument();
  });
});

describe("<Movies />の関数テスト", () => {
  describe("movieRatingClass 関数のテスト", () => {
    const movieRatingClass = (rating) =>
      rating >= 9 ? "good" : rating >= 7 ? "soso" : "bad";

    it("評価が9以上なら 'good' を返す", () => {
      expect(movieRatingClass(9)).toBe("good");
      expect(movieRatingClass(10)).toBe("good");
    });

    it("評価が7以上9未満なら 'soso' を返す", () => {
      expect(movieRatingClass(7)).toBe("soso");
      expect(movieRatingClass(8.9)).toBe("soso");
    });

    it("評価が7未満なら 'bad' を返す", () => {
      expect(movieRatingClass(6.9)).toBe("bad");
      expect(movieRatingClass(0)).toBe("bad");
    });
  });

  describe("useEffect の動作テスト", () => {
    beforeEach(() => {
      jest.spyOn(global, "fetch").mockResolvedValue({
        json: jest.fn().mockResolvedValue({
          data: {
            movies: [
              {
                id: 1,
                title: "Inception",
                year: 2010,
                rating: 8.8,
                genres: ["Action", "Adventure", "Sci-Fi"],
                runtime: 148,
                synopsis: "A thief who steals corporate secrets...",
                url: "https://example.com/inception",
                large_cover_image: "https://example.com/inception.jpg",
              },
            ],
          },
        }),
      });
    });

    afterEach(() => {
      jest.restoreAllMocks();
    });

    it("APIから映画データを取得し、表示する", async () => {
      await act(async () => {
        render(<Movies />);
      });

      await waitFor(() => {
        expect(screen.getByText("Inception (2010)")).toBeInTheDocument();
      });
    });

    it("映画データが表示される際に `movieRatingClass` が適切に適用される", async () => {
      await act(async () => {
        render(<Movies />);
      });

      await waitFor(() => {
        const ratingElement = screen.getByText("8.8");
        expect(ratingElement).toHaveClass("soso");
      });
    });
  });
});

describe("<News />のレンダリングテスト", () => {
  beforeEach(() => {
    jest.spyOn(global, "fetch").mockResolvedValue({
      json: jest.fn().mockResolvedValue([
        {
          id: 42644834,
          title: "My Amazon TV Now Unmutes Itself During Prime Video Commercial Breaks",
          user: "MourYother",
          url: "https://old.reddit.com/r/mildlyinfuriating/comments/1hx5wkq/my_amazon_tv_now_unmutes_itself_during_prime/"
        },
        {
          id: 42620001,
          title: "Nvidia releases its own brand of world models",
          user: "bariscan",
          url: "https://techcrunch.com/2025/01/06/nvidia-releases-its-own-brand-of-world-models/"
        }
      ])
    });
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("ニュースデータが正しく表示されること", async () => {
    await act(async () => {
      render(<News />);
    });

    await waitFor(() => {
      expect(screen.getByText("My Amazon TV Now Unmutes Itself During Prime Video Commercial Breaks")).toBeInTheDocument();
      expect(screen.getByText("Nvidia releases its own brand of world models")).toBeInTheDocument();
    });

    expect(screen.getByText("My Amazon TV Now Unmutes Itself During Prime Video Commercial Breaks").closest("a")).toHaveAttribute(
      "href",
      "https://old.reddit.com/r/mildlyinfuriating/comments/1hx5wkq/my_amazon_tv_now_unmutes_itself_during_prime/"
    );

    expect(screen.getByText(/\/ID: MourYother/)).toBeInTheDocument();
    expect(screen.getByText(/\/ID: bariscan/)).toBeInTheDocument();
  });
});

describe("<Todos />のテスト", () => {
  beforeEach(() => {
    jest.spyOn(global, "fetch").mockResolvedValue({
      json: jest.fn().mockResolvedValue([
        { id: 1, userId: 1, title: "Test Todo 1", completed: false },
        { id: 2, userId: 1, title: "Test Todo 2", completed: true },
      ]),
    });
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("初期レンダリングとデータ取得のテスト", async () => {
    await act(async () => {
      render(<Todos />);
    });
  
    await waitFor(() => {
      expect(screen.getByText(/Test Todo 1/)).toBeInTheDocument();
      expect(screen.getByText(/Test Todo 2/)).toBeInTheDocument();
    }, { timeout: 5000 });
  });

  it("新しいTodoの追加テスト", async () => {
    await act(async () => {
      render(<Todos />);
    });

    const input = screen.getByPlaceholderText("New Todo");
    const sendButton = screen.getByText("SEND");

    fireEvent.change(input, { target: { value: "New Test Todo" } });
    fireEvent.click(sendButton);

    await waitFor(() => {
      expect(screen.queryByText(/New Test Todo/)).toBeInTheDocument();
    });
  });

  it("Todoの削除テスト", async () => {
    await act(async () => {
      render(<Todos />);
    });
  
    await waitFor(() => {
      const deleteButtons = screen.queryAllByRole('button', { name: '❌' });
      expect(deleteButtons.length).toBeGreaterThan(0);
      fireEvent.click(deleteButtons[0]);
    });
  
    await waitFor(() => {
      expect(screen.queryByText(/Test Todo 1/)).not.toBeInTheDocument();
    });
  });

  it("Todoの完了状態の変更テスト", async () => {
    await act(async () => {
      render(<Todos />);
    });
  
    await waitFor(() => {
      const todoTitle = screen.queryByText(/Test Todo 1/);
      expect(todoTitle).toBeInTheDocument();
      fireEvent.click(todoTitle);
    });
  
    await waitFor(() => {
      const completedTodos = screen.queryAllByText("DONE ✅");
      expect(completedTodos.length).toBe(2); // 元々1つ完了済み + 新たに完了したもの
    });
  });
});