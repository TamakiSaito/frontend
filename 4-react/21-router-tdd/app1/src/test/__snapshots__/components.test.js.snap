// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Count />のテスト Snapshot test 1`] = `
<div>
  <button>
    +
  </button>
  <button>
    -
  </button>
  <div>
    Count 
    0
  </div>
</div>
`;

exports[`<Count />のレンダリングテスト Snapshot test 1`] = `
<div>
  <button>
    +
  </button>
  <button>
    -
  </button>
  <div>
    Count 
    0
  </div>
</div>
`;

exports[`<Movies />のレンダリングテスト Snapshot test 1`] = `
<div>
  <div>
    <h1>
      Movies
    </h1>
    <div />
  </div>
</div>
`;
